gazelle/python/parser.go

```diff
@@ -35,23 +36,19 @@ var (
 	parserStdin  io.WriteCloser
 	parserStdout io.Reader
 	parserMutex  sync.Mutex
+	//go:embed parse.py
+	parser     []byte
+	parserPath = path.Join(os.TempDir(), "parse.py")
 )
 
 func startParserProcess(ctx context.Context) {
-	rfiles, err := runfiles.New()
-	if err != nil {
-		log.Printf("failed to create a runfiles object: %v\n", err)
+	// "python -c" doesn't like parse.py for some reason, possibly due to the
+	// thread pool. So we need to write the code to a tmp file and execute it.
+	if err := os.WriteFile(parserPath, parser, 0644); err != nil {
+		log.Printf("cannot write %q: %s", parserPath, err.Error())
 		os.Exit(1)
 	}
-
-	parseScriptRunfile, err := rfiles.Rlocation("rules_python_gazelle_plugin/python/parse")
-	if err != nil {
-		log.Printf("failed to initialize parser: %v\n", err)
-		os.Exit(1)
-	}
-
-	cmd := exec.CommandContext(ctx, parseScriptRunfile)
-	cmd.Env = append(os.Environ(), rfiles.Env()...)
+	cmd := exec.CommandContext(ctx, "python3", parserPath)
```

Comment:
This is using system Python, right? What would be the way to use the hermetic toolchain?

Reply:
> This is using system Python, right?
Yeah, this only preserves the status quo. The old script from `rfiles.Rlocation("rules_python_gazelle_plugin/python/parse")` has this line at the beginning:
```
#!/usr/bin/env python3
```
So it's calling system Python too, although rules_python may let people feel it's using the hermetic Python toolchain.

> What would be the way to use the hermetic toolchain?
I have some idea to do it, but it's probably better to separate that into a different PR.

Reply:
The shebang there calls the system python, but then it launches the actual Python file `parse.py` using the toolchain included via the `py_binary`. I think some of this confusion stems from #691.

This PR makes the system Python dependency more present, because now the `std_module.py` will use the Python version that is on the system as opposed to the one configured by the `py_binary` if I understand things correctly.

Reply:
Oh, interesting. I didn't know that. Thanks for pointing out. Now I embed the python zip file instead, which comes with a Python interpreter. So it only relies in system python to locate that interpreter.